### ISSUE TITLE
No way to opt out of using the VERSION file

### ISSUE THREAD

**Issue by rdeago** [Buildvana/Buildvana.Sdk#160]
## Current version

1.0.0-alpha.20

## Description

The `VERSION` file was born as a temporary, quick-and-dirty fix. By now it's definitely overstayed its welcome... by years!

The problem is that I cannot explore other options (for example [`Nerdbank.GitVersioning`](https://github.com/dotnet/Nerdbank.GitVersioning), but there are more out there) because there's no way to opt out of the use of the `VERSION` file.

## Configuration

Happens on any configuration.

## Regression?

Not a regression.

## Other information

Just not parsing the `VERSION` file if it doesn't exist is not feasible for two reasons:

- The name is configurable. If a user defines `VersionFileName` in a project, they probably prefer the SDK to issue a diagnostic if it doesn't exist instead of silently skipping it; and even if they use the default name `VERSION` the reasoning is the same.
- Other versioning systems could potentially require a file of the same name, thus also reactivating the parsing logic as by (black) magic.

A `UseVersionFile` property, defaulting to `true` for compatibility, looks like a possible solution. Mentioning a feature in a project that doesn't use it looks weird (that was also the reasoning behind changing the default for `UseJetBrainsAnnotations` to `false` in 1.0.0-alpha.20); however, once a suitable versioning system is found, `VERSION` file parsing is bound to be replaced in its entirety.

In other words, I'm proposing a temporary fix to a problem caused by another temporary fix. Cringeworthy as it is, at least it's a step towards getting rid of both of them.
